globus_automate_client/models.py

```diff
@@ -0,0 +1,387 @@
+    root_validator,
+    validator,
+)
+
+
+def enforce_jsonpath(v: t.Optional[str]):
+    if v is not None and not v.startswith("$."):
+        raise ValueError(f'"{v}" must be a JSONPath and start with "$."')
+    return v
+
+
+def check_jsonpath_syntax_in_dict(item: t.Optional[dict]):
+    if item is None:
+        return item
+
+    for k, v in item.items():
+        if k.endswith(".$") and (not isinstance(v, str) or not v.startswith("$.")):
+            raise ValueError(
+                f'Key "{k}" indicates its value will be a JSONPath, however its value "{v}" is not a JSONPath'
+            )
+        elif not k.endswith(".$") and (isinstance(v, str) and v.startswith("$.")):
+            raise ValueError(
+                f'Value "{v}" is a JSONPath, however its key "{k}" does not indicate a JSONPath'
+            )
+        elif isinstance(v, dict):
+            check_jsonpath_syntax_in_dict(v)
+    return item
+
+
+class StateType(str, enum.Enum):
+    Pass = "Pass"
+    Action = "Action"
+    Wait = "Wait"
+    Choice = "Choice"
+    ExpressionEval = "ExpressionEval"
+    Fail = "Fail"
+
+
+class Catchers(BaseModel):
+    ErrorEquals: t.List[str] = Field(..., min_items=1)
+    Next: str
+    ResultPath: t.Optional[str]
+
+    _enforce_jsonpath = validator("ResultPath", allow_reuse=True)(enforce_jsonpath)
+
+
+class ChoiceRule(BaseModel):
+    Variable: str
+    StringEquals: t.Optional[str]
+    StringEqualsPath: t.Optional[str]
+    StringLessThan: t.Optional[str]
+    StringLessThanPath: t.Optional[str]
+    StringGreaterThan: t.Optional[str]
+    StringGreaterThanPath: t.Optional[str]
+    StringLessThanEquals: t.Optional[str]
+    StringLessThanEqualsPath: t.Optional[str]
+    StringGreaterThanEquals: t.Optional[str]
+    StringGreaterThanEqualsPath: t.Optional[str]
+    StringMatches: t.Optional[str]
+    NumericEquals: t.Optional[str]
+    NumericEqualsPath: t.Optional[str]
+    NumericLessThan: t.Optional[str]
+    NumericLessThanPath: t.Optional[str]
+    NumericGreaterThan: t.Optional[str]
+    NumericGreaterThanPath: t.Optional[str]
+    NumericLessThanEquals: t.Optional[str]
+    NumericLessThanEqualsPath: t.Optional[str]
+    NumericGreaterThanEquals: t.Optional[str]
+    NumericGreaterThanEqualsPath: t.Optional[str]
+    BooleanEquals: t.Optional[str]
+    BooleanEqualsPath: t.Optional[str]
+    TimestampEquals: t.Optional[str]
+    TimestampEqualsPath: t.Optional[str]
+    TimestampLessThan: t.Optional[str]
+    TimestampLessThanPath: t.Optional[str]
+    TimestampGreaterThan: t.Optional[str]
+    TimestampGreaterThanPath: t.Optional[str]
+    TimestampLessThanEquals: t.Optional[str]
+    TimestampLessThanEqualsPath: t.Optional[str]
+    TimestampGreaterThanEquals: t.Optional[str]
+    TimestampGreaterThanEqualsPath: t.Optional[str]
+    IsNull: t.Optional[str]
+    IsPresent: t.Optional[str]
+    IsNumeric: t.Optional[str]
+    IsString: t.Optional[str]
+    IsBoolean: t.Optional[str]
+    IsTimestamp: t.Optional[str]
```

Comment:
These items must be booleans.

Reply:
Good catch! Added a commit that fixes this and adds this to the tests.